src/components/Camera/Buttons/TakePhoto.tsx

```diff
@@ -21,12 +22,39 @@ interface Props {
 const TakePhoto = ( {
   takePhoto,
   disabled,
-  showPrediction
+  showPrediction,
+  debounceTime = 400,
+  preventMultipleTaps = true
 }: Props ) => {
   const { t } = useTranslation( );
 
   const borderClass = "border-[2px] rounded-full h-[64px] w-[64px]";
 
+  const [isProcessing, setIsProcessing] = useState( false );
+  const onPressRef = useRef( takePhoto );
+
+  useEffect( ( ) => {
+    onPressRef.current = takePhoto;
+  }, [takePhoto] );
+  const handleTakePhoto = ( event?: GestureResponderEvent ) => {
+    if ( !preventMultipleTaps ) {
+      onPressRef.current( event );
+      return;
+    }
+
+    if ( isProcessing ) return;
+
+    setIsProcessing( true );
+
+    onPressRef.current( event );
+
+    setTimeout( ( ) => {
+      setIsProcessing( false );
+    }, debounceTime );
+  };
+
+  const isDisabled = disabled || ( preventMultipleTaps && isProcessing );
+
   return (
     <Pressable
```

Comment:
Being a tiny bit forward thinking here, can we move this logic into a SharedComponent, maybe called `PressableWithDebounce`? I'm assuming a few things here:

1) we'll want this logic on other buttons throughout the app (maybe even in the camera, on the gallery button for example)
2) it would be nice to be able to quickly swap out a `Pressable` for a `PressableWithDebounce` in case we run into any timing issues with this (here or elsewhere)
3) it's nicer to standardize the debounce time in a single location, rather than making `TakePhoto` responsible for receiving props and knowing how long to debounce